Improve margin settings (and add explicit setting to set the `selected` window coloured border width)

jtaala opened the pull request:
Fixes #817.

See https://github.com/paperwm/PaperWM/issues/817#issuecomment-2042684866 for details.

Allows independent setting of margins and gaps (without the strange behaviour described in #817).

Adds a setting to user preferences which allows setting directly the `selected` window border "thickness":

[Screencast from 2024-04-08 23-51-21.webm](https://github.com/paperwm/PaperWM/assets/30424662/2382e4b2-87fb-437b-b0f1-60edad237af7)

[Screencast from 2024-04-09 08-26-01.webm](https://github.com/paperwm/PaperWM/assets/30424662/b1039dc0-0c7f-4b0d-91e2-2f066a138044)

jtaala committed:
Merge branch 'develop' into release

jtaala committed:
Merge branch 'develop' into release

jtaala committed:
Merge branch 'develop' into release

jtaala committed:
Merge branch 'develop' into release

jtaala committed:
Merge branch 'develop' into release

jtaala committed:
Merge branch 'develop' into release

jtaala committed:
Added `selection-border-size` preference which sets the selected window
border size.

jtaala committed:
Implemented setting to directly set the selection border size.

jtaala committed:
Removed now deprecated code.

jtaala committed:
Added setting to prefs.

jtaala committed:
Merge branch 'develop' into release

smichel17 reviewed (changes requested):
This approach acts weirdly if you set the border size to nonzero and all margins and gaps to 0.

I have some other thoughts I'll leave in #817 when I find the energy, but wanted to get this in before merge.

jtaala committed:
Merge branch 'release' into improve-margin-settings

jtaala commented:
> This approach acts weirdly if you set the border size to nonzero and all margins and gaps to 0.
> 
> I have some other thoughts I'll leave in #817 when I find the energy, but wanted to get this in before merge.

Thanks @smichel17.  I'd say it's _predictably_ weird!  But yes, this corner case doesn't quite make sense from a visual pov.

We could set the actual implementation to not let the border size be larger than margins/gaps which should resolve this case.

jtaala committed:
Updated comment re focus mode restoring.

jtaala commented:
> We could set the actual implementation to not let the border size be larger than margins/gaps which should resolve this case.

@smichel17, see implementation of this in the latest commit:

[Screencast from 2024-04-09 08-26-01.webm](https://github.com/paperwm/PaperWM/assets/30424662/b1039dc0-0c7f-4b0d-91e2-2f066a138044)

jtaala committed:
Keep border size appropriate to other margins/gaps.

jtaala committed:
Removed bottom margin from impacting selection size since some window
(like terminals) don't take full height - and users might want to set
bottom margin to 0 while keeping selection.
